pkg/controller/controller.go

```diff
@@ -426,19 +436,22 @@ func (ctrl *resizeController) resizeVolume(
 	newSize, fsResizeRequired, err := ctrl.resizer.Resize(pv, requestSize)
 
 	if err != nil {
-		klog.Errorf("Resize volume %q by resizer %q failed: %v", pv.Name, ctrl.name, err)
 		// if this error was a in-use error then it must be tracked so as we don't retry without
 		// first verifying if volume is in-use
 		if inUseError(err) {
 			ctrl.usedPVCs.addPVCWithInUseError(pvc)
 		}
-		return newSize, fsResizeRequired, fmt.Errorf("resize volume %s failed: %v", pv.Name, err)
+		return newSize, fsResizeRequired, fmt.Errorf("Resize volume %q by resizer %q failed: %v", pv.Name, ctrl.name, err)
 	}
 	klog.V(4).Infof("Resize volume succeeded for volume %q, start to update PV's capacity", pv.Name)
+	updatedPV, err := util.UpdatePVCapacity(pv, newSize, ctrl.kubeClient)
 
-	if err := util.UpdatePVCapacity(pv, newSize, ctrl.kubeClient); err != nil {
-		klog.Errorf("Update capacity of PV %q to %s failed: %v", pv.Name, newSize.String(), err)
-		return newSize, fsResizeRequired, err
+	if err != nil {
+		return newSize, fsResizeRequired, fmt.Errorf("Update capacity of PV %q to %s failed: %v", pv.Name, newSize.String(), err)
+	}
+	err = ctrl.volumes.Update(updatedPV)
```

Comment:
I was thinking to do the cache update in the util method itself, but I guess we don't want to pass `ctrl` in. Can we create another helper function here?

Reply:
made a new helper function that updates the PV with api-server and updates local cache.

Reply:
We do have now helpers functions on controller type which do the same. They patch the object and then they update the cache. Functions - `updatePVCapacity`, `markPVCAsFSResizeRequired` and `markPVCResizeFinished` both update the object with apiserver and then update the object in the cache.

Did you wanted both patching and updating in cache to be done within util functions which aren't part of controller type? In that case - we will have to pass controller instance to those functions and in which case these functions may as well be on controller type (which is what I have done).

Reply:
I was thinking about doing that but that means `PatchPVCStatus` function can't really belong to util package because it has to also update the controller's cache. I hear what you are saying but I am not sure if it is worth doing given the size of resize controller which has pretty small surface area of PV and PVC updates.